### PR TITLE
Secondary Nav 2.0

### PR DESCRIPTION
**Pull request by gfunk1230**
**Preview:**

https://user-images.githubusercontent.com/6556242/167238587-aa67b701-e4f1-41d9-a135-678dea8f5910.mov

**TODO:**

- Need to add an active link when on the designated route. Thought I could just ask @mcolella14 for help on this one. Figured that can speed things up so that I can work on the homepage. 
- Maybe write tests but TBH not sure if we need it. I am open to suggestions though.

### CODE REVIEW

**Review by mcolella14 — COMMENTED** [review]
There's only a few blocking things: 

- The taxonomy names need to be updated.

- On mobile, I can't seem to scroll when the nav takes up more than the height of the viewport. Is this something you experienced?

Other than that, could we move most of the styling into a separate file? I see there's a small `styles.ts` file, but there's still a ton of styling happening in the components themselves and it's bulking up the files.

**Comment by mcolella14** [comment]
I put together a small POC using `next/router` to show how we can pass the current path down to the nav. You can then conditionally style based on whether that path matches the link's slug. This is is what the `secondarynavnew/index.tsx` file would look:
```typescript
import { useRouter } from 'next/router';
import MobileView from './mobile';
import DesktopView from './desktop';
import { useEffect, useState } from 'react';

const SecondaryNav = () => {
    const [activePath, setActivePath] = useState<string>();
    const { isReady, asPath } = useRouter();
    useEffect(() => {
        if (isReady) {
            setActivePath(asPath);
        }
    }, [isReady]);

    return (
        <>
            <MobileView />
            <DesktopView activePath={activePath} />
        </>
    );
};

export default SecondaryNav;

```

File: src/components/seconardynavnew/mobile.tsx [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+                        />
+                    )}
+                    {isOpen && (
+                        <SystemIcon
+                            sx={plusOrMinusStylesForDropDowns}
+                            name={ESystemIconNames.MINUS}
+                            size="small"
+                            color="success"
+                        />
+                    )}
```
[the remark this answers is not in the document]
Comment (gfunk1230) [reply src/components/seconardynavnew/mobile.tsx:193]: @mcolella14 is this the same as this suggestion https://github.com/mongodb/devcenter/pull/63#discussion_r868222938

[the remark this answers is not in the document]
Comment (mcolella14) [reply src/components/seconardynavnew/mobile.tsx:55]: @gfunk1230 yeah it's the same thing, sorry should have linked to that comment

File: src/data/secondary-nav.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                name: 'Expertise Levels',
+                slug: '',
+                all: '',
+                path: '/all-topics',
+                dropDownItems: [
+                    {
+                        name: 'Introductory',
+                        slug: '/expertise-levels/introductory',
+                    },
+                    { name: 'Expert', slug: '/expertise-levels/expert' },
```
Comment (mcolella14) [inline src/data/secondary-nav.ts:105]: These pages don't exist. I'm not sure what the intention was from design, so it's worth asking Mike, but you can just direct these to `/search?expertiseLevel =Introductory` now that the search is merged.

**Review by mcolella14 — APPROVED** [review]

**Pull request merged**